### PR TITLE
add ability to warn or fail a test if a seed is fixed

### PR DESCRIPTION
**Pull request by osi**
introduce a `FixedSeedMode` that can be set globally (via properties)
or on an individual test. this allows control of using the `seed`
argument on a property, as it can be undesirable to accidentally commit
a value in this field after reproducing a failure.

fix for #138

### CODE REVIEW

File: engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java [hunk 1]
```diff
@@ -55,7 +55,17 @@ public CheckedProperty(
...
+		try {
+			effectiveConfiguration = configurationWithEffectiveSeed();
+		} catch (FailOnFixedSeedException failOnFixedSeedException) {
+			return PropertyCheckResult.failed(
+					configuration.getStereotype(), propertyName, 0, 0,
+					configuration.getSeed(), configuration.getGenerationMode(),
+					configuration.getEdgeCasesMode(), 0, 0,
+					null, null, failOnFixedSeedException
+			);
+		}
```
Comment (osi) [inline engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java:68]: This catch block feels funny. I can't merge it with the next though due to it needing the actual `effectiveConfiguration`

Comment (jlink) [reply engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java:68]: Cannot see a more elegant solution either atm. Sometimes bad looking code has to stay until s.o. suddenly has an epiphany.

**Comment by jlink** [comment]
@osi Do you think it's ready for merging?

**Comment by osi** [comment]
I believe it's ready. The CI failure seems like a permission issue of some sort.

**Comment by jlink** [comment]
Yes, CI is on me. 

Von meinem iPad gesendet

**Review by jlink — CHANGES_REQUESTED** [review]
If you don't find the time I can do the few changes myself.

**Comment by osi** [comment]
> If you don't find the time I can do the few changes myself.

I'll take care of them in the next few days. I appreciate the suggestions!

File: engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java [hunk 2]
```diff
@@ -55,7 +55,17 @@ public CheckedProperty(
 	}
 
 	public PropertyCheckResult check(Reporting[] reporting) {
-		PropertyConfiguration effectiveConfiguration = configurationWithEffectiveSeed();
+		PropertyConfiguration effectiveConfiguration;
+		try {
```
[the remark this answers is not in the document]
Comment (osi) [reply engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java:59]: The problem with moving into that method is that the try-catch block translates the exception to a `PropertyCheckResult`. 

I extracted the handling of the exception into a helper method, as the same work is done twice. perhaps that's sufficient for now?

**Comment by jlink** [comment]
Many thanks. Going to merge it.

**Pull request merged**